app/admin/issue.rb

```diff
@@ -5,62 +5,82 @@
+  scope(:closed) { |scope| AdminIssues.for(current_admin).merge(IssuesClosed.for).merge(scope) }
+
+  order_by(:type) do |order_clause|
+    "#{order_clause.to_sql}, id #{order_clause.order}"
+  end
 
   index do
-    column :issue_type_name, class: :left do |issue|
-      link_to issue.issue_type_name, issue_path(id: issue.id)
-    end
+    column :issue_type_name, class: :left, sortable: :type, &:view_link_with_name
     column :level_name
     column :objects_links
     column :assigned_to
-    actions
+    column :status do |issue|
+      status_tag(issue.status_name, class: issue.status)
+    end
+    actions defaults: false, &:view_link
   end
 
   show do
-    attributes_table do
-      row :issue_type_name
-      row :description
-      row :objects_links
-      row :role_name
-      row :level_name
-      row :assigned_to
-      row :created_at
-      row :updated_at
-      row :fixed_at
+    panel t("census.issues.title") do
+      render "show"
     end
 
     show_table context: self, title: t("census.issues.information"), table: resource.information
+    show_table context: self, title: t("census.issues.fix_information"), table: resource.fix_information
+  end
+
+  form title: I18n.t("census.issues.close.action"), decorate: true do |f|
+    render "issues/#{resource.issue_type}/form", context: self, f: f, issue: resource
   end
+  sidebar :issue, partial: "issues/show", only: :edit
 
   action_item :assign_me, only: :show do
-    unless resource.fixed_at || resource.assigned_to_id == current_admin.person_id
+    unless resource.closed? || resource.assigned_to_id == current_admin.person_id
       link_to t("census.issues.assign_me"), assign_me_issue_path, method: :patch, data: { confirm: t("census.messages.sure_question") }
     end
   end
 
-  action_item :mark_as_fixed, only: :show do
-    link_to t("census.issues.mark_as_fixed"), mark_as_fixed_issue_path, method: :patch, data: { confirm: t("census.messages.sure_question") } unless resource.fixed_at
-  end
-
   member_action :assign_me, method: :patch do
     Issues::AssignIssue.call(issue: resource, admin: current_admin)
     redirect_back(fallback_location: issues_path)
   end
 
-  member_action :mark_as_fixed, method: :patch do
-    Issues::FixedIssue.call(issue: resource, admin: current_admin)
-    redirect_back(fallback_location: issues_path)
-  end
-
   controller do
     before_action only: :show do
       Issues::ReadIssue.call(issue: resource, admin: current_admin)
     end
+
+    def edit
+      redirect_back(fallback_location: issues_path, error: t("census.issues.action_message.already_closed")) && return if resource.closed?
+      super
+    end
+
+    def update
+      issue = resource
+      issue.assign_attributes permitted_params[:issue]
+
+      Issues::FixIssue.call(issue: issue, admin: current_admin) do
+        on(:invalid) { render :edit }
+        on(:error) do
+          flash[:error] = t("census.issues.action_message.error")
```

Comment:
I think this needs to be `flash.now[:error]` or it won't be rendered. We'll find out when this code path is tested :)

Reply:
Thanks! I've checked this in all the project code and improved tests to check it in my last commits.